Userland/Libraries/LibAudio/FlacLoader.cpp

```diff
@@ -354,7 +354,15 @@ ErrorOr<Vector<FixedArray<Sample>>, LoaderError> FlacLoaderPlugin::load_chunks(s
 
     size_t sample_index = 0;
 
-    while (!m_stream->is_eof() && sample_index < samples_to_read) {
+    // FIXME: To prevent this workaround that allows the test for a FLAC file with an unknown amount of samples to pass,
+    //  is_eof() should return true at the same time as tell() == size() returns true
```

Comment:
@timschumi explains is_eof() behavior on discord sometimes

Reply:
Too often. :^)

I have a suspicion for what's causing this, I'll start a bisect later.

Reply:
I see some comments about when EOF should actually be set, and that is after the final read that returns 0, if we know we're about to return 0, EOF shouldn't be set yet.


That was the behaviour I noticed when poking around `Stream::read_until_filled()` for this particular test failure, and does make sense. But the problem is, at least with this FLAC infrastructure, by the time EOF is actually set it is too late and we're already in the middle of trying to read another frame that doesn't exist.

Maybe the PluginLoader infrastructure needs to have a way to bail out and assume/account for this EOF behaviour?

Reply:
We don't know that we are about to return 0, we'd have to explicitly ask each and every time that we'd want to get an accurate `is_eof` reading (thrice even, because there is no POSIX API to check the EOF condition in particular, so we'd have to seek multiple times to figure out where we are and where the end of the file is). We could offer that as `is_eof_slow`, but that would only work for seekable streams.

The unfortunate thing is that this can affect any user of `Stream`, as long as it was nice enough to read until the exact end of the data, and as long as it can only know if the file ends by trying to read the next set of data.

I'll finish up the bisect (which should be soon, because I have a few likely candidates in mind), and then I'll check whether we can substitute the whole thing with something like `poll`.